### ISSUE TITLE
The validation generator is resetting the model?

### ISSUE THREAD

**Issue by WilliamOu** [fchollet/keras#6082]
Hey, guys. I am training a regression model by fit_generator. I find something wired. 

In the first epoch of training, the loss decreased from about 30 to about 1; but at the beginning of next epoch, the loss goes back to 30 again. Then the training process continues from that 30. I trained the model for 10 epochs, and found that at the end of every epoch, the loss does not decrease, they more seems like random values. Can anybody help me to find the bug. I have been struggling with this for two days. 
The code is here

```
from keras.callbacks import TensorBoard, ModelCheckpoint, EarlyStopping, CSVLogger
from keras.optimizers import Adam
from HistoryUtlisCopy2 import CreateModel, Supplier, Evaluator, SupplierSteps, EvaluatorSteps
import os


# ***→→→ History Experiment ←←←***
Scale = 0.25
TimeLen = 10
print("The time length is: %d" % TimeLen)
BatchSize = 16

model = CreateModel(TimeLen=TimeLen, Scale=Scale)
print("The number of parameters is: %d" % model.count_params())
model.compile(loss='mean_squared_error', optimizer=Adam(lr=0.00002))

FileDirt = "./CheckPoint" + str(TimeLen)
if not os.path.exists(FileDirt):
    os.mkdir(FileDirt)

saver = ModelCheckpoint(filepath=FileDirt + "/Weights-{epoch:02d}-{loss:.4f}.hdf5",
                        monitor='loss',
                        save_best_only=False,
                        save_weights_only=True,
                        period=1)
stopper = EarlyStopping(monitor='loss', patience=10)
logger = CSVLogger(filename=FileDirt + '/TrainLog.csv')
boarder = TensorBoard(log_dir=FileDirt + '/Logs')
callbacks = [saver, stopper, logger, boarder]

model.fit_generator(generator=Supplier(BatchSize=BatchSize, TimeLen=TimeLen, Scale=Scale),
                    steps_per_epoch=1000,#SupplierSteps(BatchSize=BatchSize, TimeLen=TimeLen),
                    validation_data=Evaluator(BatchSize=BatchSize, TimeLen=TimeLen, Scale=Scale),
                    validation_steps=10,#EvaluatorSteps(BatchSize=BatchSize, TimeLen=TimeLen),
                    epochs=10,
                    verbose=1,
                    callbacks=callbacks)
```

**Comment by WilliamOu** [comment]
I checked again my code. I found the problem is with the validation part of my code. I am using a generator to provide data for the validation. 

If I comment out this two lines in the fit_generator code which are the 4th and 5th lines to the end, the training process is correct. With this two lines. the code does not work well. I am planning to change to another method to provide data for  the validation.

Well it is still great if someone can provide some ideas about this code.
